addon/components/ember-collection.js

```diff
@@ -187,11 +188,11 @@ export default Ember.Component.extend({
           set(cell, 'hidden', false);
           cellMap[itemKey] = cell;
         } else {
-          set(cell, 'hidden', true);
-          set(cell, 'style', 'height: 0; display: none;');
+          cellsToDelete.push(cell);
```

Comment:
It's been so long but keeping the cell around might have some performance wins if the collection shrinks, but then grows again ( you don't have to pay to re-create the DOM the second time). Rather than deleting the cell could we instead just set `item` to `null`?